request.go

```diff
@@ -999,7 +999,115 @@ func putTextprotoReader(r *textproto.Reader) {
 	textprotoReaderPool.Put(r)
 }
 
+// ReadRequest reads and parses an incoming request from b.
```

Comment:
this is gigantic file. i think theres a ton of refactoring needed here.

Reply:
oh is this a library? while not use a package manager?

Reply:
This file is actually a file taken out of Go's net/http library. I wanted to use it b/c this file dissects a request string into a request struct object which is then passed on to the routing, handling, and transferring steps.

Did not think to use a package manager b/c pasting this code into my project aided in my understanding of how Go's http server worked.